sdk/tables/azure-data-tables/tests/test_retry.py

```diff
@@ -38,6 +40,52 @@ def send(self, request, **kwargs):
         raise ServiceResponseError(timeout_error, error=timeout_error)
 
 
+class FailoverRetryTransport(RequestsTransport):
+    """Transport to attempt to raise on first request but allow requests to secondary location.
+    To simulate a failover scenario, pass the exception into the `failover` keyword arg on the method
+    you wish to test, e.g.:
+    client.get_entity("foo", "bar", failover=ServiceRequestError("Attempting to force failover"))
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._already_raised = False
+
+    def send(self, request, **kwargs):
+        # secondary endpoint is not allowed on DELETE operations
+        if request.method != "DELETE":
+            if self._already_raised:
+                assert "-secondary" in request.url
+            else:
+                assert "-secondary" not in request.url
+        failover_on_error = kwargs.pop("failover", None)
+        if failover_on_error and not self._already_raised:
+            self._already_raised = True
+            raise failover_on_error
+
+        self._already_raised = False
+        return super().send(request, **kwargs)
+
+
+class SecondaryFailoverRetryTransport(RequestsTransport):
```

Comment:
Rather than define a whole new transport here, you could simply configure this in the transport above using a constructor param:
```python
class FailoverRetryTransport(RequestsTransport):
    """Transport to attempt to raise on first request but allow requests to secondary location.
    To simulate a failover scenario, pass the exception into the `failover` keyword arg on the method
    you wish to test, e.g.:
    client.get_entity("foo", "bar", failover=ServiceRequestError("Attempting to force failover"))
    """

    def __init__(self, location_mode='secondary', *args, **kwargs):
        super().__init__(*args, **kwargs)
        self._location_mode = location_mode
        self._already_raised = False

        # secondary endpoint is not allowed on DELETE operations
        if request.method != "DELETE":
            if self._already_raised or self._location_mode == 'secondary':
                assert "-secondary" in request.url
            else:
                assert "-secondary" not in request.url
```

Reply:
Merged it into `FailoverRetryTransport`